zenodo/modules/pages/utils.py

```diff
@@ -66,6 +70,33 @@ def format_request_email_body(context):
     template = current_app.config['PAGES_EMAIL_BODY_TEMPLATE'],
     return render_template_to_string(template, context)
 
+
+def check_attachment_size(attachments):
+    """Function to check size of attachment within limits.
+
+    :param attachments: Attachments files.
+    :returns: Boolean.
+    """
+    if check_attachment_size(attachments) == 0:
+        return False
+    if attachments:
+        if 'Content-Length' in request.headers:
```

Comment:
There's also `request.content_length`. Using these properties of `request` offer sometimes some safety and "free" parsing